openwisp_monitoring/device/tests/test_models.py

```diff
@@ -578,6 +578,73 @@ def _create_env(self):
         )
         return dm, ping, load, process_count
 
+    def test_disabling_critical_check(self):
+        Check = load_model('check', 'Check')
+        dm, ping, load, process_count = self._create_env()
+        ping_check_instance = Check.objects.create(
+            name='Ping Check',
+            check_type='ping',
+            content_object=dm.device,
+            params={},
+        )
+        dm.update_status('ok')
+        with catch_signal(health_status_changed) as handler:
+            ping_check_instance.is_active = False
+            ping_check_instance.save()
+        self.assertEqual(handler.call_count, 1)
+        call_args = handler.call_args[1]
+        self.assertEqual(call_args['instance'], dm)
+        self.assertEqual(call_args['status'], 'unknown')
+        dm.refresh_from_db()
+        self.assertEqual(dm.status, 'unknown')
+        with self.subTest(
+            "Ensure status doesn't change on saving active critical check"
+        ):
+            ping_check_instance.is_active = True
+            ping_check_instance.save()
+            dm.refresh_from_db()
+            self.assertEqual(dm.status, 'ok')
```

Comment:
This is so wrong @praptisharma28, the description you just wrote above says "Ensure status doesn't change on saving active critical check", but the status here is changing! This assertion must be `self.assertEqual(dm.status, 'unknown')`